tests/ui/unsized-locals/extern-parameter-issue-123887.rs

```diff
@@ -0,0 +1,12 @@
+// https://github.com/rust-lang/rust/issues/123887
+// Do not ICE on unsized extern parameter
+//@ compile-flags: -Clink-dead-code --emit=link
+#![feature(extern_types, unsized_fn_params)]
+
+extern "C" {
+    type ExternType;
+}
+
+fn f(_: ExternType) {} //~ ERROR the size for values of type `ExternType` cannot be known at compilation time
```

Comment:
What about `Wrapper<T: ?Sized>(T)` and `fn f(_: Wrapper<ExternType>)`? Does that ICE?

Reply:
Ah, good point.  Yes it did.  In a04fe4d84f5acc2c0693907453d86a363458cc64 I've copied the code from the triggering assertion to get to the tail type, and am using that instead.